src/info/manifest.c

```diff
@@ -0,0 +1,1986 @@
+
+                MEM_CONTEXT_TEMP_RESET(1000);
+            }
+        }
+        MEM_CONTEXT_TEMP_END();
+    }
+
+    // -----------------------------------------------------------------------------------------------------------------------------
+    if (infoSaveSection(infoSaveData, MANIFEST_SECTION_TARGET_LINK_DEFAULT_STR, sectionNext))
+    {
+        if (lstSize(manifest->linkList) > 0)
+        {
+            infoSaveValue(
+                infoSaveData, MANIFEST_SECTION_TARGET_LINK_DEFAULT_STR, MANIFEST_KEY_GROUP_STR,
+                jsonFromVar(saveData->linkGroupDefault, 0));
+            infoSaveValue(
+                infoSaveData, MANIFEST_SECTION_TARGET_LINK_DEFAULT_STR, MANIFEST_KEY_USER_STR,
+                jsonFromVar(saveData->linkUserDefault, 0));
+        }
+    }
+
+    // -----------------------------------------------------------------------------------------------------------------------------
+    if (infoSaveSection(infoSaveData, MANIFEST_SECTION_TARGET_PATH_STR, sectionNext))
+    {
+        MEM_CONTEXT_TEMP_RESET_BEGIN()
+        {
+            for (unsigned int pathIdx = 0; pathIdx < manifestPathTotal(manifest); pathIdx++)
+            {
+                const ManifestPath *path = manifestPath(manifest, pathIdx);
+                KeyValue *pathKv = kvNew();
+
+                if (!varEq(manifestOwnerVar(path->group), saveData->pathGroupDefault))
+                    kvPut(pathKv, MANIFEST_KEY_GROUP_VAR, manifestOwnerVar(path->group));
+
+                if (path->mode != saveData->pathModeDefault)
+                    kvPut(pathKv, MANIFEST_KEY_MODE_VAR, VARSTR(strNewFmt("%04o", path->mode)));
+
+                if (!varEq(manifestOwnerVar(path->user), saveData->pathUserDefault))
+                    kvPut(pathKv, MANIFEST_KEY_USER_VAR, manifestOwnerVar(path->user));
+
+                infoSaveValue(infoSaveData, MANIFEST_SECTION_TARGET_PATH_STR, path->name, jsonFromKv(pathKv, 0));
+
+                MEM_CONTEXT_TEMP_RESET(1000);
+            }
+        }
+        MEM_CONTEXT_TEMP_END();
+    }
+
+    // -----------------------------------------------------------------------------------------------------------------------------
+    if (infoSaveSection(infoSaveData, MANIFEST_SECTION_TARGET_PATH_DEFAULT_STR, sectionNext))
+    {
+        infoSaveValue(
+            infoSaveData, MANIFEST_SECTION_TARGET_PATH_DEFAULT_STR, MANIFEST_KEY_GROUP_STR,
+            jsonFromVar(saveData->pathGroupDefault, 0));
+        infoSaveValue(
+            infoSaveData, MANIFEST_SECTION_TARGET_PATH_DEFAULT_STR, MANIFEST_KEY_MODE_STR,
+            jsonFromStr(strNewFmt("%04o", saveData->pathModeDefault)));
+        infoSaveValue(
+            infoSaveData, MANIFEST_SECTION_TARGET_PATH_DEFAULT_STR, MANIFEST_KEY_USER_STR,
+            jsonFromVar(saveData->pathUserDefault, 0));
+    }
+
+    FUNCTION_TEST_RETURN_VOID();
+}
+
+void
+manifestSave(Manifest *this, IoWrite *write)
```

Comment:
So where is infoManifestSaveFile (i.e. where we would add the cipherBlock filter and save the file and a copy (or not)? I'm guessing we'd have one infoManifestSaveFile or maybe infoManifestSaveFile and infoManifestSaveFileCopy (for aborted backups)?

Reply:
Saving a manifest to file is a complex multi-step process that is done in backup.  A general function would not have enough info to do it safely.